ocaml/typing/typeopt.ml

```diff
@@ -22,9 +22,6 @@ open Asttypes
 open Typedtree
 open Lambda
 
-(* CR layout v2: Error infrastructure was added to this module just to support
```

Comment:
I'm surprised to see this removed. I thought the end goal was not to have any errors in typeopt. Anything that would error here would be caught elsewhere. It's true that we'll still have to look at layouts to decide whether or not to call `value_kind`, but no cases in that match should error, I think.

Reply:
I think, unfortunately, we will always have the possibility that the checks here fail due to `any` (perhaps as a result of a missing cmi), and we will want to error in that case.